### PR TITLE
Override web component sdk configs

### PR DESCRIPTION
**Pull request by asafshen**
## Related Issues
https://github.com/descope/etc/issues/5319

## Related PRs

https://github.com/descope/descope-js/pull/356

## Description
Override web component sdk configs so that the hooks running before will run in the same manner

### CODE REVIEW

File: src/components/Descope.tsx [hunk 1]
```diff
@@ -10,12 +10,27 @@ import React, {
...
-	module.default.sdkConfigOverrides = { baseHeaders };
+	module.default.sdkConfigOverrides = {
+		baseHeaders,
+		// the inner web-component will not persist the tokens
+		// So the global SDK hooks will handle the tokens according to the SDK configuration
+		persistTokens: false,
+		hooks: {
+			get beforeRequest() {
+				// Done we can use the same hooks in the web-component according to the SDK configuration
+				return getGlobalSdk().httpClient.hooks.beforeRequest;
```
Comment (nirgur) [inline src/components/Descope.tsx:27]: I find the name `globalSdk` a bit confusing, I think `sdkInstance` was better 😄

File: src/components/Descope.tsx [hunk 2]
```diff
@@ -10,12 +10,27 @@ import React, {
...
 	// we want to override the web-component base headers so we can tell that is was used via the React SDK
-	module.default.sdkConfigOverrides = { baseHeaders };
+	module.default.sdkConfigOverrides = {
+		baseHeaders,
+		// the inner web-component will not persist the tokens
+		// So the global SDK hooks will handle the tokens according to the SDK configuration
+		persistTokens: false,
+		hooks: {
+			get beforeRequest() {
+				// Done we can use the same hooks in the web-component according to the SDK configuration
```
Comment (nirgur) [inline src/components/Descope.tsx:26]: This comment is not so clear to me

File: src/components/Descope.tsx [hunk 3]
```diff
@@ -10,12 +10,27 @@ import React, {
...
+				return getGlobalSdk().httpClient.hooks.beforeRequest;
+			},
+			set beforeRequest(_) {
```
Comment (nirgur) [inline src/components/Descope.tsx:29]: Can we just remove it?

**Review by nirgur — COMMENTED** [review]
Looks good
Added a few small comments

File: src/components/Descope.tsx (hunk 1 above)
Comment (asafshen) [reply src/components/Descope.tsx:27]: I think the name `global` tells more about the fact that this sdk is on global scope, and I think that the term `instance` doesn't add any information about this variable

I think that if we expose this object, lets say to the user (developer), I think that `globalSdk` signals the proper message

lmk what you think

File: src/components/Descope.tsx (hunk 3 above)
Comment (asafshen) [reply src/components/Descope.tsx:29]: I thought I added it for a reason, but it indeed seems redundant, removing 👍

File: src/components/Descope.tsx (hunk 2 above)
Comment (asafshen) [reply src/components/Descope.tsx]: improved, let me know what you think

**Review by nirgur — APPROVED** [review]

**Pull request merged**